node/exercises/scrab-bag.js

```diff
@@ -0,0 +1,43 @@
+const colors = require('colors')
+const quitCommand = require('../helpers/quitCommand')
+const scrabbleLetters = require('../constants/scrabbleLetters')
+
+const letters = Object.keys(scrabbleLetters)
+let scrabBag = Object.assign(scrabbleLetters)
+
+console.log('Enter the tiles that have already been played.'.green)
+
+process.stdin.resume();
+process.stdin.setEncoding('utf8');
+process.stdin.on('data', function(input){
+  if(input.indexOf('/') === 0) {
+    quitCommand(input, process)
+  } else {
+    input = input.replace(/\W|\d/g, '').toUpperCase().split('')
+    console.log(input)
+
+    input.forEach(letter => {
+      scrabBag[letter].count = scrabBag[letter].count - 1
+    })
+
+    let remainingLetters = {}
+
+    letters.forEach(letter => {
+      let count = scrabBag[letter].count;
+      if(remainingLetters.hasOwnProperty(count)){
+        remainingLetters[count].push(letter)
+      } else {
+        remainingLetters[count] = [letter]
+      }
+    })
+
+    let remainingLetterKeys = Object.keys(remainingLetters)
+
+    console.log('Letters remaining in the bag:'.green)
+    remainingLetterKeys.forEach(key => {
+      let currentLetters = []
+      console.log(colors.green(key + ': ' + remainingLetters[key].join(', ')))
+    })
+    process.disconnect()
+  }
+})
```

Comment:
Next time try defining your second argument in its own function and then calling it in your parent function argument declaration.

To up your game with your code use the single responsibility principle. Each function should be responsible for one output. Define extra functions outside the main function and then call them inside as parameters

https://en.wikipedia.org/wiki/Single_responsibility_principle